From 4752e641be4815994488d6d6910b01bc011fcdd7 Mon Sep 17 00:00:00 2001
From: Davor Bertovic <davor@losinj.com>
Date: Fri, 30 Dec 2011 21:18:47 +0100
Subject: [PATCH] fix for 32bit colors and refresh in ICS

---
 drivers/video/msm/msm_fb.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/video/msm/msm_fb.c b/drivers/video/msm/msm_fb.c
index 9116459..76d8f31 100644
--- a/drivers/video/msm/msm_fb.c
+++ b/drivers/video/msm/msm_fb.c
@@ -1114,7 +1114,7 @@ static ssize_t debug_read(struct file *file, char __user *buf, size_t count,
 };
 #endif
 
-#define BITS_PER_PIXEL_DEF 16
+#define BITS_PER_PIXEL_DEF 32
 
 static void setup_fb_info(struct msmfb_info *msmfb)
 {
@@ -1140,7 +1140,7 @@ static void setup_fb_info(struct msmfb_info *msmfb)
 	fb_info->var.yres_virtual = msmfb->yres * 2;
 	fb_info->var.bits_per_pixel = BITS_PER_PIXEL_DEF;
 	fb_info->var.accel_flags = 0;
-
+	fb_info->var.reserved[5] = 60;
 	fb_info->var.yoffset = 0;
 
 	if (msmfb->panel->caps & MSMFB_CAP_PARTIAL_UPDATES) {
-- 
1.7.5.4

